src/arch/mod.rs

```diff
@@ -31,24 +31,55 @@ lazy_static! {
 mod x64;
 pub use x64::*;
 
-pub fn union_size(symbols: &[Symbol]) -> Result<SIZE_T, &'static str> {
-    symbols
-        .iter()
-        .map(|symbol| symbol.ctype.sizeof())
-        // max of member sizes
-        .try_fold(1, |n, size| Ok(max(n, size?)))
-}
-
-pub fn struct_size(symbols: &[Symbol]) -> Result<SIZE_T, &'static str> {
-    symbols.iter().try_fold(0, |offset, symbol| {
-        Ok(Type::next_offset(offset, &symbol.ctype)?)
-    })
-}
+impl StructType {
+    /// Get the offset of the given struct member.
+    pub fn struct_offset(&self, member: super::InternedStr) -> u64 {
```

Comment:
Unrelated, but can you make this `pub(crate)`? Originally I just made everything `pub` but I've been trying to avoid that because a) it makes the API harder to use and b) the compiler doesn't warn you about unused code


```suggestion
    pub(crate) fn struct_offset(&self, member: super::InternedStr) -> u64 {
```

Reply:
Ditto for the other functions you changed.